connaisseur/validate.py

```diff
@@ -121,20 +120,20 @@ def process_chain_of_trust(host: str, image: Image, req_delegations: list):
         else:
             raise NotFoundException("could not find any delegations in trust data.")
 
-    # get a list from all `targets` fields from `targets.json` + delegation roles or
-    # just the delegation roles, should there be required delegation according to the
-    # policy
+    # if certain delegations are required, then only take the targets fields of those
+    # delegation JSONs. otherwise take the targets field of the targets JSON, as long as
+    # no delegations are defined in the targets JSON. in that case the targets field of
+    # the releases JSON will be used.
     if req_delegations:
         image_targets = [
             trust_data[target_role].signed.get("targets", {})
             for target_role in req_delegations
         ]
     else:
-        image_targets = [
-            trust_data[target_role].signed.get("targets", {})
-            for target_role in trust_data
-            if re.match("targets(/[^/\\s]+)?", target_role)
-        ]
+        if trust_data["targets"].has_delegations():
+            image_targets = [trust_data["targets/releases"].signed.get("targets", {})]
+        else:
+            image_targets = [trust_data["targets"].signed.get("targets", {})]
```

Comment:
```
targets_key = "targets/releases" if trust_data["targets"].has_delegations() else "targets"
image_targets = [trust_data[targets_key].signed.get("targets", {})]
```

This removes the code duplication (`.signed.get(...)`). (The point is not that it has less lines of code.)